trieste/acquisition/function/active_learning.py

```diff
@@ -245,3 +245,153 @@ def acquisition(x: TensorType) -> TensorType:
         return criterion
 
     return acquisition
+
+
+class IntegratedVarianceReduction(SingleModelAcquisitionBuilder):
+    """
+    Builder for the reduction of the integral of the predicted variance over the search
+    space given a batch of query points.
+    """
+
+    def __init__(
+        self, integration_points: TensorType, threshold: Optional[Sequence[float]] = None
+    ) -> None:
+        """
+        :param integration_points: set of points to integrate the prediction variance over.
+        :raise ValueError (or InvalidArgumentError): If ``integration_points`` does not have
+                exactly two dimensions, or if the first dimension has size 0.
+        """
+        self._integration_points = integration_points
+        self._threshold = threshold
+
+    def __repr__(self) -> str:
+        """"""
+        return f"IntegratedVarianceReduction(threshold={self._threshold!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The integral of the predictive variance.
+        """
+
+        return integrated_variance_reduction(model, self._integration_points, self._threshold)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+class integrated_variance_reduction(AcquisitionFunctionClass):
+    """
+    The reduction of the average of the predicted variance over the integration points
+    (a.k.a. Integrated Means Square Error or IMSE criterion).
+    See :cite:`Picheny2010` for details.
+
+    If no threshold is provided, the goal is to learn a globally accurate model, and
+    the original variance is used. Otherwise, learning is 'targeted' towards regions
+    where the GP is close to particular values, and the variance is weighted by the
+    posterior GP pdf evaluated at the threshold (if a single value is given) or by the
+    probability that the GP posterior belongs to the interval between the 2 thresholds.
```

Comment:
It would be good to mention that the function can do batches of points, a brief explanation about updated variance (models need to be able to do this...), and computational costs, recommended batch size etc